File: frontend/types/Student.ts
// Student interface matching Faculty_System database schema
export interface Student {
  Stu_ID: number;
  Stu_SSN: string;
  FName: string;
  LName: string;
  Email: string;
  Username: string;
  Password: string;
  Gender: 'M' | 'F';
  Level: 1 | 2 | 3 | 4;
  DOB: string;
  Age?: number; // Computed field
  CGPA: number;
  Dept_ID: number;
  // Related data
  phones?: StudentPhone[];
  addresses?: StudentAddress[];
  enrollments?: StudentEnrollment[];
}

export interface StudentPhone {
  Stu_ID: number;
  Phone_NUM: string;
}

export interface StudentAddress {
  Stu_ID: number;
  City: string;
  Street: string;
}

export interface CreateStudentRequest {
  Stu_ID: number;
  Stu_SSN: string;
  FName: string;
  LName: string;
  Email: string;
  Username: string;
  Password: string;
  Gender: 'M' | 'F';
  Level: 1 | 2 | 3 | 4;
  DOB: string;
  CGPA: number;
  Dept_ID: number;
}

export interface UpdateStudentRequest extends Partial<CreateStudentRequest> {}

export interface StudentEnrollment {
  Stu_ID: number;
  Crs_ID: number;
  Grade?: number; // 0-100
  Year: string; // e.g., '2024-25'
}

File: api/Models/Student.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{
    [Table("Student", Schema = "Faculty")]
    public class Student
    {
        [Key]
        public int Stu_ID { get; set; }

        [Required]
        [StringLength(14)]
        public string Stu_SSN { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string FName { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string LName { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Email { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string Username { get; set; } = string.Empty;

        [Required]
        [StringLength(255)]
        public string Password { get; set; } = string.Empty;

        [Required]
        [StringLength(1)]
        public string Gender { get; set; } = string.Empty;

        [Required]
        [Range(1, 4)]
        public int Level { get; set; }

        public DateTime DOB { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public int? Age { get; set; }

        [Required]
        [Column(TypeName = "decimal(3,2)")]
        public decimal CGPA { get; set; }

        [Required]
        public int Dept_ID { get; set; }

        // Navigation properties
        [ForeignKey("Dept_ID")]
        public virtual Department? Department { get; set; }

        public virtual ICollection<StudentPhone> Phones { get; set; } = new List<StudentPhone>();
        public virtual ICollection<StudentAddress> Addresses { get; set; } = new List<StudentAddress>();
        public virtual ICollection<Enrollement> Enrollments { get; set; } = new List<Enrollement>();
    }
}

